Application/v1/viewModels/CurrencyViewModel.go

```diff
@@ -36,6 +37,34 @@ func (c *CurrencyViewModel) MarshalJSON() (b []byte, e error) {
 	})
 }
 
+func (c *CurrencyViewModel) validateAmount() bool{
+	if c.Amount < 0 {
+		c.validation.AddMessage(validation.GenMessage("amount", "Is negative"))
+		return false
+	}
+	return true
+}
+
+func (c *CurrencyViewModel) validateCurrentCurrency() bool {
+	for i, cc := range Domain.Currencies {
+		if cc == Domain.Currencies[i]{
```

Comment:
это условие всегда верно, тебе надо 
```suggestion
		if cc == c.CurrentCurrency {
```